command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -321,6 +349,57 @@ export class Sequence {
 }
 
 //helper functions
+
+// @ts-ignore : Used in generated code
+function A(absoluteTime: String | Temporal.Instant): GeneratedCommands {
+  return commandsWithTimeValue(absoluteTime, TimingTypes.ABSOLUTE);
+}
+
+// @ts-ignore : Used in generated code
+function R(duration: String | Temporal.Duration): GeneratedCommands {
+  return commandsWithTimeValue(duration, TimingTypes.COMMAND_RELATIVE);
+}
+
+// @ts-ignore : Used in generated code
+function E(duration: String | Temporal.Duration): GeneratedCommands {
+  return commandsWithTimeValue(duration, TimingTypes.EPOCH_RELATIVE);
+}
+
+function commandsWithTimeValue(
```

Comment:
We can better type this as
```typescript
function commandsWithTimeValue<T extends TimingTypes>(
  timeValue: T extends TimingTypes.Absolute ? string | Temporal.Instant : string | Temporal.Duration,
  timeType: T
): GeneratedCommands
```

This allows us to enforce that absolute uses strings or instants and relative uses strings or durations